back/src/modules/game/Game.class.ts

```diff
@@ -0,0 +1,228 @@
+import { WebsocketsService } from '../websockets/websockets.service';
+import {
+	convertStateToSendable,
+	GameStatus,
+	getDefaultGameState,
+	IBall,
+	IGameState,
+	IKeyEvent,
+	IPlayer,
+	IPlayerInfos,
+	IPosition,
+	IRect,
+} from './Game.interfaces';
+
+export class Game {
+	private websocketsService: WebsocketsService;
+
+	private _player1: IPlayer;
+	private _player2: IPlayer;
+	private _status: GameStatus = GameStatus.STARTING;
+
+	private _startCounter: number = 10;
+
+	private _gameState: IGameState;
+
+	constructor(
+		player1: IPlayer,
+		player2: IPlayer,
+		websocketsService: WebsocketsService,
+	) {
+		this._player1 = player1;
+		this._player2 = player2;
+		this.websocketsService = websocketsService;
+		this._gameState = getDefaultGameState(player1, player2);
+	}
+
+	async start() {
+		while (this._startCounter > 0) {
+			//TODO: Change to 1000 ms
+			await this._wait(100);
+			this._startCounter--;
+			this._sendToPlayers('match-starting', { time: this._startCounter });
+		}
+		this._sendToPlayers('match-starting', { time: this._startCounter });
+		this._status = GameStatus.PLAYING;
+		this._game();
+	}
+
+	getPlayer(userId: number): IPlayer | null {
+		if (!this._player1 || !this._player2) return null;
+		if (this._player1.user.id === userId) {
+			return this._player1;
+		} else if (this._player2.user.id === userId) {
+			return this._player2;
+		} else {
+			return null;
+		}
+	}
+
+	processInput(userId: number, data: IKeyEvent) {
+		const player = this.getPlayer(userId);
+		if (!player) return;
+		if (data.action === 'press') {
+			if (player.user.id === this._player1.user.id) {
+				this._gameState.player1.event = data.direction;
+			}
+			if (player.user.id === this._player2.user.id) {
+				this._gameState.player2.event = data.direction;
+			}
```

Comment:
C'est moche, 2 noms similaires pour 2 trucs différents (player et this._gameState.player(1/2))